experiment/simulation/js/fa.js

```diff
@@ -0,0 +1,333 @@
+        this.inputPoints = [];
+        this.outputPoints = [];
+        this.CoutIsConnected = false;
+        this.sumIsConnected = false;
+        this.component = '<div class="drag-drop FullAdder" id=' + this.id + ' style="width:100px;height:100px;"></div>';
+    }
+
+    // Adds element to the circuit board, adds event listeners and generates its endpoints.
+    registerComponent(workingArea,x = 0, y = 0) {
+        const parent = document.getElementById(workingArea);
+        parent.insertAdjacentHTML('beforeend', this.component);
+        document.getElementById(this.id).style.left = x + "px";
+        document.getElementById(this.id).style.top = y + "px";
+
+        const el = document.getElementById(this.id);
+        el.addEventListener('contextmenu', function (ev) {
+            ev.preventDefault();
+                let left = ev.pageX - document.getScroll()[0];
+                let top = ev.pageY - document.getScroll()[1];
+                const origin = {
+                    left: left,
+                    top: top
+                };
+                setPosition(origin);
+            window.selectedComponent = this.id;
+            window.componentType = "fullAdder";
+            // deleteElement(this.id);
+            return false;
+        }, false);
+
+        fullAdder[this.id] = this;
+        registerGate(this.id, this);
+    }
+
+    // Sets values of the inputs and outputs of the full adder
+    setA0(A0) {
+        this.a0 = A0;
+    }
+
+    setB0(B0) {
+        this.b0 = B0;
+    }
+
+    setCin(Cin) {
+        this.Cin = Cin;
+    }
+
+    setSum(Sum) {
+        this.sum = Sum;
+    }
+
+    setCout(Cout) {
+        this.Cout = Cout;
+    }
+
+    // adds input endpoints points to the list of input points
+    addInputPoints(input) {
+        this.inputPoints.push(input);
+    }
+
+    // Adds the output endpoints to the list of output points
+    addOutputPoints(output) {
+        this.outputPoints.push(output);
+    }
+
+    // Generates the output of the full adder
+    generateOutput() {
+        const aXorb = (getOutputFA(this.a0[0], this.a0[1]) && !getOutputFA(this.b0[0], this.b0[1])) || (!getOutputFA(this.a0[0], this.a0[1]) && getOutputFA(this.b0[0], this.b0[1]));
+        this.Cout = (getOutputFA(this.a0[0], this.a0[1]) && getOutputFA(this.b0[0], this.b0[1])) || (getOutputFA(this.Cin[0], this.Cin[1]) && aXorb);
+        this.sum = (getOutputFA(this.Cin[0], this.Cin[1]) && !aXorb) || (!(getOutputFA(this.Cin[0], this.Cin[1])) && aXorb);
+    }
+
+    // Sets the output enpoint of the full adder as connected
+    setConnected(val, pos) {
+        console.log(val, pos);
+        if (pos === "Carry") {
+            this.CoutIsConnected = val;
+        }
+        else if (pos === "Sum") {
+            this.sumIsConnected = val;
+        }
+    }
+
+}
+
+// Add a full adder to the circuit board
+function addFA() {
+    let fA = new FullAdder();
+    fA.registerComponent("working-area");
+}
+
+window.AddFA = addFA;

```

Comment:
Please use camelCase identifiers for functions

Reply:
done